cmd/skopeo/signing.go

```diff
@@ -88,3 +89,137 @@ var standaloneVerifyCmd = cli.Command{
 	ArgsUsage: "MANIFEST DOCKER-REFERENCE KEY-FINGERPRINT SIGNATURE",
 	Action:    standaloneVerify,
 }
+
+func addSignature(context *cli.Context) error {
+	if len(context.Args()) != 2 {
+		return errors.New("Usage: skopeo add-signature reference key-fingerprint")
+	}
+	ref := context.Args()[0]
+	fingerprint := context.Args()[1]
+
+	imageRef, err := transports.ParseImageName(ref)
+	if err != nil {
+		return fmt.Errorf("Error parsing %s: %v", ref, err)
+	}
+	name := transports.ImageName(imageRef)
+
+	// TODO: the signed named reference is always relative to the transport.  Is this the right value?
+	signReference := imageRef.StringWithinTransport()
```

Comment:
No; it would be `imageRef.DockerReference` _if it is not `nil`_.  The `docker-reference` field in a signature is using the `containers/image/docker/reference` semantics, it is not transport-specific.

If `imageRef.DockerReference` is `nil`, the user would have to specify the signed identity explicitly (probably as an option, not as a mandatory argument). Or it can be left unimplemented for now.